aw-server/src/endpoints/settings.rs

```diff
@@ -0,0 +1,67 @@
+use std::sync::MutexGuard;
+use rocket_contrib::json::Json;
+use rocket::http::Status;
+use rocket::State;
+use crate::endpoints::ServerState;
+
+use aw_datastore::{Datastore, DatastoreError};
+
+fn parse_key(key: String) -> Result<String, Status> {
+    if key.len() >= 128 { return Err(Status::BadRequest) }
+    else { Ok(key + ".settings")  }
```

Comment:
This is confusing to me for two reasons:

 - The `settings` should be prepended, not appended.
 - I'm not sure if the API should restrict access to other namespaces, but maybe you're right that it should.

Reply:
Hm yes great point about it needing to be prepended, not appended haha. That's the intended behaviour, somehow it slipped my mind.

I think I'm convinced with @johan-bjareholt 's arguments about making the API's purpose clear and restricted, but creating the sql support in a way that theoretically could support new endpoints in the future is for the best.